pkg/testing/fixture_install.go

```diff
@@ -170,6 +170,38 @@ func (f *Fixture) Install(ctx context.Context, installOpts *InstallOpts, opts ..
 	c := client.New(client.WithAddress(socketPath))
 	f.setClient(c)
 
+	f.t.Cleanup(func() {
+		if f.t.Failed() {
+			procs := getProcesses(f.t, `.*`)
+			dir, err := findProjectRoot(f.caller)
+			if err != nil {
+				f.t.Logf("failed to dump process; failed to find project root: %s", err)
+				return
+			}
+
+			// Sub-test names are separated by "/" characters which are not valid filenames on Linux.
+			sanitizedTestName := strings.ReplaceAll(f.t.Name(), "/", "-")
+
+			filePath := filepath.Join(dir, "build", "diagnostics", fmt.Sprintf("TEST-%s-ProcessDump.json", sanitizedTestName))
```

Comment:
Did you manage to see a build where this artifact was attached? Can you link it here?

Reply:
It's in `build/diagnostics` , like in this failed run https://buildkite.com/elastic/elastic-agent/builds/7359#018dccc8-2c82-4845-9aa7-6440420fb850

Reply:
Great! Thanks!

Reply:
We probably want to put the test architecture in the artifact name. You'll be able to tell windows from linux by the contents, but not arm vs x86 linux.

Diagnostics have the same naming problem, but the variables file tells you the host arch when you open them.

Reply:
added os and arch in 1f9348a8147471c5397c6c58725d71299b58f224